File: backend/src/test/java/com/skewwhiffy/auraltester/cucumber/ClefSteps.java
package com.skewwhiffy.auraltester.cucumber;

import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;
import lombok.AllArgsConstructor;
import lombok.val;
import org.springframework.http.HttpStatus;

import static org.assertj.core.api.AssertionsForClassTypes.assertThat;

@AllArgsConstructor
public class ClefSteps {
    private final RestRequests restRequests;

    @When("I request a(n) {} clef")
    public void when_requestClef(String clef) {
        restRequests.makeGetRequest("/api/clef?clef=" + clef);
    }

    @Then("I get a clef")
    public void then_getClef() {
        val response = restRequests.getLastResponse();
        assertThat(response.getStatus()).isEqualTo(HttpStatus.OK.value());
        val responseMap = restRequests.getLastResponseAsMap();
        assertThat(responseMap.get("abc")).isInstanceOf(String.class);
    }
}

File: backend.old/src/test/kotlin/com/skewwhiffy/auraltester/cucumber/ClefSteps.kt
package com.skewwhiffy.auraltester.cucumber

import io.cucumber.java.en.Then
import io.cucumber.java.en.When
import org.assertj.core.api.Assertions.assertThat
import org.springframework.http.HttpStatus

class ClefSteps(private val restRequests: RestRequests) {
    @When("I request a(n) {} clef")
    fun when_requestClef(clef: String) {
        restRequests.makeGetRequest("/api/clef?clef=$clef")
    }

    @Then("I get a clef")
    fun then_getClef() {
        val response = restRequests.lastResponse
        assertThat(response.status).isEqualTo(HttpStatus.OK.value())
        val responseMap = restRequests.lastResponseAsMap
        assertThat(responseMap["abc"]).isInstanceOf(String::class.java)
    }
}
